examples/linear_model/plot_lasso_and_elasticnet.py

```diff
@@ -3,95 +3,168 @@
+# `alpha` and `l1_ratio`. To select their optimal value we used an
+# :class:`~sklearn.linear_model.ElasticNetCV`, not shown here to keep the
+# example simple.
 
 from sklearn.linear_model import ElasticNet
 
-enet = ElasticNet(alpha=alpha, l1_ratio=0.7)
+t0 = time()
+enet = ElasticNet(l1_ratio=0.8, **common_params).fit(X_train, y_train)
+print(f"ElasticNet fit done in {(time() - t0):.3f}s")
 
-y_pred_enet = enet.fit(X_train, y_train).predict(X_test)
+y_pred_enet = enet.predict(X_test)
 r2_score_enet = r2_score(y_test, y_pred_enet)
-print(enet)
-print("r^2 on test data : %f" % r2_score_enet)
-
+print(f"ElasticNet r^2 on test data : {r2_score_enet:.3f}")
 
 # %%
-# Plot
-# ---------------------------------------------------
-
-m, s, _ = plt.stem(
-    np.where(enet.coef_)[0],
-    enet.coef_[enet.coef_ != 0],
-    markerfmt="x",
-    label="Elastic net coefficients",
-)
-plt.setp([m, s], color="#2ca02c")
-m, s, _ = plt.stem(
-    np.where(lasso.coef_)[0],
-    lasso.coef_[lasso.coef_ != 0],
-    markerfmt="x",
-    label="Lasso coefficients",
-)
-plt.setp([m, s], color="#ff7f0e")
-plt.stem(
-    np.where(coef)[0],
-    coef[coef != 0],
-    label="true coefficients",
-    markerfmt="bx",
+# Plot results
+# ------------
+#
+# In this section we use a heatmap to visualize the sparsity of the true
+# and estimated coefficients of the respective linear models.
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+import pandas as pd
+from matplotlib.colors import SymLogNorm
+
+df = pd.DataFrame(
+    {
+        "True weights": true_weights,
+        "Lasso": lasso.coef_,
+        "ARDRegression": ard.coef_,
+        "ElasticNet": enet.coef_,
+    }
 )
 
-plt.legend(loc="best")
+plt.figure(figsize=(10, 6))
+ax = sns.heatmap(
+    df.T,
+    norm=SymLogNorm(linthresh=10e-4, vmin=-80, vmax=80),
+    cbar_kws={"label": "coefficients' values"},
+    cmap="seismic_r",
+)
+plt.ylabel("linear model")
+plt.xlabel("coefficients")
 plt.title(
-    "Lasso $R^2$: %.3f, Elastic Net $R^2$: %.3f" % (r2_score_lasso, r2_score_enet)
+    f"Models' coefficients\nLasso $R^2$: {r2_score_lasso:.3f}, "
+    f"ARD $R^2$: {r2_score_ard:.3f}, "
+    f"ElasticNet $R^2$: {r2_score_enet:.3f}"
 )
-plt.show()
+plt.tight_layout()
+
+# %%
+# Conclusions
+# -----------
+#
+# :class:`~sklearn.linear_model.Lasso` is known to recover sparse data very well
+# but does not perform well with highly correlated features.
+#
+# :class:`~sklearn.linear_model.ARDRegression` is better when handling gaussian
+# noise, which translates in a slightly better score than Lasso, but is still
+# unable to handle correlated features and requires a large amount of time due
+# to fitting a prior.
+#
+# :class:`~sklearn.linear_model.ElasticNet` does not have a problem with
+# correlated features but is unable to recover the true generating process of
+# sparse data as good as Lasso. In this case, this is the model with the best
```

Comment:
Basically, this is untrue because it depends on the `l1_ratio`. I assume that the story should be said differently.

```
# :class:`~sklearn.linear_model.ElasticNet` is a middle ground between
# :class:`~sklearn.linear_model.Lasso` and :class:`~sklearn.linear_model.Ridge`.
# It imposes sparsity on the coefficients and shrinks the coefficient to zero.
# Thus, if correlated features all contribute to the target, the model will
# still be able to affect a weight to all of them. However, the behaviour is
# controlled by the two hyperparameters `l1_ratio` and `alpha` which can make the
# model difficult to tune in practice.
```

Reply:
> which can make the model difficult to tune in practice.

I don't understand this point, wouldn't cross-validating do the trick efficiently?

Reply:
Since this is the combination of 2 parameters, you can have several models that will perform equally. If the generative model is sparse and with correlated features, the Lasso will only recover partially the support while a L2/L1 (`ElasticNet`) potentially should recover the true generative model because the mix of both will affect equal weights to the correlated feature. However, in terms of predictions, both models can be consistent and provide the same output. The `GridSearchCV` will select the best performer statistically but it does mean that this is the model the closest to the true generative model given random score fluctuation.

Reply:
Thank you for the clarification. I assume you made a typo at the end of your comment?
> but it does [**not**] mean that this is the model the closest to the true generative model

Reply:
indeed ;)